src/components/brickModal/BrickModal.tsx

```diff
@@ -167,25 +182,26 @@ const ItemRenderer = ({
 
 // Need to provide a key because we reorder elements on search
 // See https://react-window.vercel.app/#/api/FixedSizeGrid
-type ItemKeyInput = {
+type ItemKeyInput<T extends IBrick> = {
   columnIndex: number;
-  data: ItemType;
+  data: ItemType<T>;
   rowIndex: number;
 };
 // Here, we use the brick id as the key, which is the "value" prop on the search result option
-function itemKey({
+function itemKey<T extends IBrick>({
   columnIndex,
-  data: { searchResults },
+  data: { brickResults },
   rowIndex,
-}: ItemKeyInput): RegistryId | number {
+}: ItemKeyInput<T>): RegistryId | number {
   const resultIndex = getFlatArrayIndex(rowIndex, columnIndex);
   // Number of bricks for the last Grid row could be less than the number of columns
   // Returning the index here, ItemRenderer will render an empty cell
-  if (resultIndex >= searchResults.length) {
+  if (resultIndex >= brickResults.length) {
     return resultIndex;
   }
 
-  return searchResults[resultIndex]?.value;
+  // eslint-disable-next-line security/detect-object-injection -- index is a number
+  return brickResults[resultIndex]?.value;
```

Comment:
Let's use [`[].at()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at) to access dynamic indexes in arrays
```suggestion
  return brickResults.at(resultIndex)?.value;
```

Reply:
I didn't know about `[].at()` before. It looks like it makes lookups at the end of an array easier using negative index numbers? What's the reason you want me to use it here? To avoid the lint warning about object injection? I would hesitate to do that just in this one spot, considering the vast number of places in the codebase where we use square-brackets and suppress the lint warning already... Would rather stick to common patterns than introduce something new in one spot.

Reply:
I'm going to merge this PR for now, but happy to keep discussing this if you think there are good reasons to maybe try and change our square bracket calls to use `at()` across the codebase. Feel free to drop a Slack post in #engineering about it if you want!